tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,419 @@
+        sub = OCP(
+            kind=constants.SUBSCRIPTION,
+            namespace=constants.OPENSHIFT_LOGGING_NAMESPACE,
+        )
+        logging_sub = sub.get().get("items")
+        if not logging_sub:
+            assert "Logging is not configured"
+
+    @system_test
+    @polarion_id("OCS-3976")
+    @skipif_no_kms
+    @skipif_ocs_version("<4.11")
+    def test_graceful_nodes_shutdown(
+        self,
+        multi_obc_lifecycle_factory,
+        nodes,
+        setup_mcg_bg_features,
+        validate_mcg_bg_features,
+        snapshot_restore_factory,
+    ):
+        """
+        Steps:
+          1) Have a cluster with FIPS, hugepages, encryption enabled
+          2) Create some resources: Create PVC (encrypt & non encrypt),
+            take snapshot and restore it into new PVCs, clone PVC
+          3) Configure OCP workloads(monitoring, registry, logging)
+          4) Create rgw kafka notifications and objects should be notified
+          5) Create normal OBCs and buckets
+          6) Perform mcg bucket replication (uni and bidirectional) and see the objects are synced.
+          7) Perform noobaa caching
+          8) Perform S3 operations on NSFS bucket on files : put, get, delete, list objects
+          9) With all the above preconditions met, follow the KCS steps for graceful shutdown of nodes
+            https://access.redhat.com/articles/5394611, and bring up the cluster back.
+            Once cluster up, there should not be seen any issues or impact on any data(No DU/DL/DC) and also
+            normal operations should work fine.
+        """
+
+        # Create normal OBCs and buckets
+        multi_obc_lifecycle_factory(num_of_obcs=20, bulk=True, measure=False)
+
+        # OCP Workloads
+        logger.info("start_ocp_workload")
+        start_ocp_workload(
+            workloads_list=["monitoring", "registry", "logging"], run_in_bg=True
+        )
+
+        # Setup MCG Features
+        logger.info(
+            "Setup MCG Features- Bucket replication,"
+            " Noobaa caching,Object expiration,"
+            "MCG NSFS,RGW kafka notification"
+        )
+        setup_mcg_bg_features(skip_any_features=["expiration", "nsfs"])
+
+        # check OSD status after graceful node shutdown
+        worker_nodes = get_nodes(node_type="worker")
+        logger.info(f"worker nodes: {worker_nodes}")
+        master_nodes = get_nodes(node_type="master")
+        logger.info("Gracefully Shutting down worker & master nodes")
+        nodes.stop_nodes(nodes=worker_nodes, force=False)
+        nodes.stop_nodes(nodes=master_nodes, force=False)
+        time.sleep(1800)
+        logger.info("Starting worker & master nodes")
+        nodes.start_nodes(nodes=worker_nodes)
+        nodes.start_nodes(nodes=master_nodes)
+
+        wait_for_cluster_connectivity(tries=400)
+        Sanity().health_check(tries=60)
+
+        self.validate_data_integrity()
+        self.validate_snapshot_restore(snapshot_restore_factory)
```

Comment:
where are we checking the below step-
PVC expansion is possible on the restored snapshot-  validate_pvc_expansion

Reply:
https://github.com/red-hat-storage/ocs-ci/pull/8233/files#diff-56d19df1f041e2768208d6a3ec334de5933efc60cdd389ec07e0332b787243fcR274

Reply:
calling validate_pvc_expansion inside validate_snapshot_restore